datasets/flwr_datasets/resplitter/divide_resplitter_test.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2023 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""DivideResplitter tests."""
+
+import unittest
+from typing import Dict, Union
+
+from parameterized import parameterized_class
+
+from datasets import Dataset, DatasetDict
+from flwr_datasets.resplitter import DivideResplitter
+
+
+@parameterized_class(
+    ("divide_config", "divide_split", "drop_remaining_splits", "split_name_to_size"),
+    [
+        # Standard config that sums to one
+        (
+            {"train_1": 0.25, "train_2": 0.75},
+            "train",
+            False,
+            {"train_1": 10, "train_2": 30, "valid": 20, "test": 40},
+        ),
+        # As the first use case but drop the remaining splits
+        (
+            {"train_1": 0.25, "train_2": 0.75},
+            "train",
+            True,
+            {"train_1": 10, "train_2": 30},
+        ),
+        # Split does not sum to 1.0
```

Comment:
```suggestion
        # Split without ratios
        (
            {"train_1": 15, "train_2": 25},
            "train",
            False,
            {"train_1": 15, "train_2": 25, "valid": 20, "test": 40},
        ),
        # Split does not sum to 1.0
```

Reply:
I'm thinking worth adding this other test that triggers a slightly different part of the `resplit` logic.

Reply:
Sure, it's now extended by the example you gave